index.js

```diff
@@ -23,8 +28,7 @@ module.exports = function (input, length) {
 
 	while (i--) {
 		var x = pathParts[i];
-
-		if (remainingLength < x.length) {
+		if (remainingLength < x.length + 1) {
```

Comment:
Nice catch, would it be possible to add your failing test to this PR to document the changed behavior, instead of creating a new PR?